ocaml/xapi/xapi_event.ml

```diff
@@ -655,18 +655,20 @@ let from_inner __context session subs from from_t timer batching =
   in
   let events_of ~kind ?(with_snapshot = true) entries acc =
     let rec go events ({table; obj; time= _} as entry) =
-      let snapshot =
-        let serialiser = Eventgen.find_get_record table in
-        if with_snapshot then
-          serialiser ~__context ~self:obj ()
+      try
+        let snapshot =
+          let serialiser = Eventgen.find_get_record table in
+          if with_snapshot then
+            serialiser ~__context ~self:obj ()
+          else
+            None
+        in
+        let event = event_of kind ?snapshot entry in
+        if Subscription.event_matches subs event then
+          event :: events
         else
-          None
-      in
-      let event = event_of kind ?snapshot entry in
-      if Subscription.event_matches subs event then
-        event :: events
-      else
-        events
+          events
+      with _ -> events
```

Comment:
Do we want to log this? Might want to add a comment why they can be ignored and don't represent a problem.

Reply:
Maybe we want to log it when the kind is not `del`, when the kind is 'del' I think it is expected that find_get_record would fail.

Reply:
Or maybe this is happening on `mod` races, haven't been able to repro this through the python API, but booting/shutting down VMs in parallel causes XenCenter to lose connection quite often